src/app/Http/Controllers/SignupController.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\User;
+use App\Models\Profile;
+use Illuminate\Http\Request;
+use Illuminate\View\View;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Http\RedirectResponse;
+use App\Http\Requests\SignupRequest;
+
+class SignupController extends Controller
+{
+
+    /**
+     * Return view user to sign up page.
+     */
+    public function showSignup(): view
+    {
+        return view('components.forms.signup');
+    }
+
+
+    /**
+     * Register a new user.
+     */
+    public function register(SignupRequest $request): View
+    {
+        $requestData = $request->all();
+        unset($requestData["password_confirmation"]);
+
+        $user = User::create([
+            'status' => 0,
+            'username' => $requestData['username'],
+            'password' => $requestData['password'],
+            'email' => $requestData['email']
+        ]);
+
+        Profile::create([
+            'user_id' => $user->id,
+        ]);
```

Comment:
You can create a service layer for this to separate the logic from the redirection. The controller must only handle requests and the service handles the data logic e.g. creating, editing, deleting etc

ref:
https://medium.com/@miladev95/service-layer-in-laravel-a4fb8e45d9ee